app/cards/[id]/charts/page.tsx

```diff
@@ -0,0 +1,54 @@
+"use client"
+import { Chart } from "@/components/chart"
+import { useParams } from "next/navigation"
+import { getCardById } from "@/domains/cards/services"
+import { CardRecord } from "@/domains/cards/types"
+import { DEFAULT_HEIGHTforAGE } from "./constant"
+
+export default function ChartItem() {
+  const { id: idParam } = useParams()
+  const id = idParam as string
+  const card = getCardById(id)
+  if (!card) {
+    return
+  }
+  const data: CardRecord[] = card.records
```

Comment:
I suggest you to rename this to `chartData`. Name `data` can not tell what is stored under this variable in context of the whole component.